xen/arch/arm/vsmc.c

```diff
@@ -93,7 +93,7 @@ static bool handle_arch(struct cpu_user_regs *regs)
     switch ( fid )
     {
     case ARM_SMCCC_VERSION_FID:
-        set_user_reg(regs, 0, ARM_SMCCC_VERSION_1_1);
+        set_user_reg(regs, 0, ARM_SMCCC_VERSION_1_2);
```

Comment:
Correct me if I am wrong but here you make Xen claim it is supporting 1.2 version but in practice all calls would still be using 1.1

Reply:
This was reviewed at https://lore.kernel.org/all/0c6d58ae-bd8d-a3d8-d8d9-2b7a5340b4dd@xen.org/

Reply:
nobody commented on that part and i do not think we should claim to support 1.2 if (or until) all calls are not following the 1.2 standard otherwise a guest might have wrong expectations.

Reply:
This is needed by the Linux kernel FF-A driver.
As far as I understand we are compliant with SMCCC version 1.2. If you see anything mandatory missing please let me know and I'll add it.

Reply:
1.2 is saying that the non used registers should be preserved so nothing needs to be changed unless we have calls using more registers. So it is ok.

Reply:
Thanks